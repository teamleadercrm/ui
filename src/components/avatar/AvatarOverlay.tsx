import React, { PureComponent } from 'react';
import theme from './theme.css';
import Box from '../box';
import Icon from '../icon';
import { IconEditSmallFilled } from '@teamleader/ui-icons';
import cx from 'classnames';

class AvatarOverlay extends PureComponent {
  render() {
    const { size } = this.props;
    const classNames = cx(
      theme['overlay-background'],
      size === 'medium' ? theme['full-height'] : theme['default-height'],
    );

    return (
      <div {...this.props} className={theme['overlay']}>
        <Box className={classNames} display="flex" justifyContent="center">
          <Icon color="neutral" tint="lightest">
            <IconEditSmallFilled />
          </Icon>
        </Box>
      </div>
    );
  }
}

export default AvatarOverlay;
