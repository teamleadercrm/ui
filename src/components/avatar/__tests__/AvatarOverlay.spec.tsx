import React from 'react';
import { render } from '@testing-library/react';
import AvatarOverlay from '../AvatarOverlay';

describe('Component - AvatarOverlay', () => {
  it('renders', async () => {
    const { asFragment } = render(<AvatarOverlay size="medium" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
