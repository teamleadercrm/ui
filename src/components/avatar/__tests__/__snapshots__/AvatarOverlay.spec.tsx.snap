// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - AvatarOverlay renders 1`] = `
<DocumentFragment>
  <div
    class="overlay"
    data-teamleader-ui="avatar-overlay"
  >
    <div
      class="box display-flex justify-content-center overlay-background full-height"
    >
      <span
        class="box align-items-center display-flex neutral lightest"
        data-teamleader-ui="icon"
      >
        <svg
          fill="currentColor"
          height="14"
          preserveAspectRatio="xMidYMid meet"
          style="opacity: 0.84; vertical-align: middle;"
          viewBox="0 0 14 14"
          width="14"
        >
          <path
            clip-rule="evenodd"
            d="M11.73 5.022L8.876 2.17l.63-.63c.393-.394.91-.59 1.426-.59a2.012 2.012 0 012.017 2.016 2.01 2.01 0 01-.59 1.426l-.63.63zm-.708.707l-5.794 5.795-3.33 1.11A.5.5 0 011.267 12l1.11-3.33L8.17 2.877l2.852 2.852z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
